doc/developer/design/20230322_pg_source_subsource_filter.md

```diff
@@ -0,0 +1,101 @@
+- Feature name: Filter PG subsources by schemas using `FOR SCHEMAS`
+- Associated:
+    - [implementation](https://github.com/MaterializeInc/materialize/pull/18220)
+    - [usability epic](https://github.com/MaterializeInc/materialize/issues/17598)
+
+# Summary
+[summary]: #summary
+
+Right now you can only ingest tables from a PG publication using an explicit
+list or simply ingesting all of them.
+
+If a publication has multiple tables of the same name, even if in different
+schemas, this makes it impossible to use the option to ingest all of the
+tables––we default to creating a subsource with the same name as the object and
+this tries to create two objects with the same name in the same transaction.
```

Comment:
I am confused about the semantics of the proposed feature. If `public` and `project` both contain a table named `foo`, does `FOR SCHEMAS (public, project)` create two subsources named `db.public.foo` and `db.project.foo`? Or is the idea that if `public` and `staging` contain `foo`, but `project` does not, the `FOR ALL SCHEMAS` lets you conveniently exclude the problematic `staging` schema without a very annoying manual list of all the tables you want in `public` and `project`?

I like the sound of the latter behavior better, but realized I wasn’t clear on which is intended!

Reply:
> ...`FOR SCHEMAS` lets you conveniently exclude the problematic `staging` schema without a very annoying manual list of all the tables you want in public and project?

This behavior is the intended result. Glad you like the sound of that better. :)